### PR TITLE
Add nodes to an existing instance

### PR DESCRIPTION
**Pull request by Wikiki**
Add new nodes based on a selector to an existing instance.

Example:
 $( '#div1' ).add( '#div2' )

### CODE REVIEW

**Comment by webpro** [comment]
What functionality are you missing from `.push()` (array) and/or e.g. `.append()` (DOM)?

**Comment by Wikiki** [comment]
Hi,

I miss the chainability from push and using `.append()` does not works. If I log the DOMtastic instance after an .append() it's still empty.
And `.push()` function doesn't return a DOMtastic instance.

Example:
```
console.log( $().add('.sortable') );           // Display a list of all matching elements
console.log( $().push('.sortable') );         // Display 1
console.log( $().append('.sortable') );     // Display an empty array
```


regards,

**Comment by webpro** [comment]
* `append` indeed doesn't work with a selector: https://domtastic.js.org/doc/#dom/index-append
* `push` works like `Array.prototype.push`
* How about a new `concat` method? Which can be used like this:

```
$(selector).concat(NodeList);
$('.first').concat($('.second'));
$().concat($('.sortable'));

```

**Comment by Wikiki** [comment]
Why not having a new `.concat()` method instead of a `.add()` one.

**Comment by webpro** [comment]
Great idea. Could you please add specs and make sure the code style matches the rest of the project (`npm run lint`)?

**Comment by Wikiki** [comment]
Maybe you can help me, it's my first time with mocha.

When I run `npm run test` on my side it's working perfectly well but when it's run by Travis CI my test does not pass. I got the error `TypeError: nodes.indexOf is not a function`

**Comment by Wikiki** [comment]
The problem is only for `npm run test:jquery`
Any idea why indexOf doesn't work with jquery-compat test ?

File: src/selector/extra.js [hunk 1]
```diff
@@ -143,3 +143,13 @@ export const siblings = function(selector) {
 export const slice = function(start, end) { // eslint-disable-line no-unused-vars
   return $([].slice.apply(this, arguments));
 };
+
+export const concat = function( elements ) {
+  const nodes = this;
+  each( elements, element => {
+    if( nodes.indexOf( element ) === -1 ) {
```
Comment (webpro) [inline src/selector/extra.js]: The jquery-compat mode has an `index` method (not `indexOf`), like jQuery itself; so you can't use this directly. Perhaps something like `[].indexOf.call(this, element)` is better here.

File: src/selector/extra.js [hunk 2]
```diff
@@ -143,3 +143,13 @@ export const siblings = function(selector) {
...
+      nodes.push( element );
+    }
+  } );
+  return $( nodes );
```
Comment (webpro) [inline src/selector/extra.js]: Please return `this`; no need to create a new instance?

**Comment by Wikiki** [comment]
Thanks for your help.
I've made the corrections.

**Comment by Wikiki** [comment]
Can I add the following code at the beginning of the method and still aligned with DOMtastic philosophy:

```
if ( typeof elements === 'string' ) {
    elements = querySelector( elements );
  }
```

to be able to concat from a selector ?

**Pull request merged**